### PR TITLE
[GenericSystem] Add velocity-only mode in fake_components 

### PR DESCRIPTION
**Pull request by livanov93**
PR info:

- Add `disable_command` flag to be able to simulate disconnected driver
- Add `prepare_command_mode_switch` and `perform_command_mode_switch` for switching between velocity-only and position-only controllers
- Add first-order integration of velocity for position calculation in velocity mode

### CODE REVIEW

File: hardware_interface/src/fake_components/generic_system.cpp [hunk 1]
```diff
@@ -278,20 +311,134 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
...
+    {
+      if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_POSITION)
+      {
+        start_modes_.push_back(hardware_interface::HW_IF_POSITION);
+      }
+      if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_VELOCITY)
+      {
+        start_modes_.push_back(hardware_interface::HW_IF_VELOCITY);
+      }
+    }
```
Comment (destogl) [inline hardware_interface/src/fake_components/generic_system.cpp:337]:
```suggestion
  hardware_interface::return_type ret_val = hardware_interface::return_type::OK;

  // Currently we don't accept ACCELERATION-only and EFFORT-only interfaces
  for (const auto & key : start_interfaces)
  {
    for (auto i = 0ul; i < info_.joints.size(); i++)
    {
      if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_ACCELERATION || 
          key == info_.joints[i].name + "/" + hardware_interface::HW_IF_EFFORT)
      {
        ret_val = false;
      }
    }
    
    return ret_val;
```

File: hardware_interface/src/fake_components/generic_system.cpp [hunk 2]
```diff
@@ -278,20 +311,134 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
...
+    std::find(start_modes_.begin(), start_modes_.end(), hardware_interface::HW_IF_VELOCITY) !=
+      start_modes_.end())
+  {
+    for (size_t i = 0; i < joint_commands_[VELOCITY_INTERFACE_INDEX].size(); ++i)
+    {
+      joint_commands_[VELOCITY_INTERFACE_INDEX][i] = 0.0;
+    }
+    velocity_controller_running_ = true;
+  }
+  return ret_val;
```
Comment (destogl) [inline hardware_interface/src/fake_components/generic_system.cpp:392]:
```suggestion
  hardware_interface::return_type ret_val = hardware_interface::return_type::OK;

// class variable
  std::vector<std::vector<bool>> used_interface_per_joint_[interface_type][joint];
  // set everything to false first
  
  for (const auto & interface : start_interfaces)
  {
    joint_name = split (interface.split("/")[0])
    interface_name = split (interface.split("/")[1])
    
    auto found_it = std::find(standard_interfaces_.begin(), standard_interfaces_.end(), interface_name)
    if (found_it != standard_interfaces_.end())
    {
       auto pos = std::dist(standard_interfaces_.begin(), found_it);
       
       auto found_joint = std::find_if(info_.joints.being(), info_.joints.end(), [&](const auto & joint) { return joint.name == joint_name; });
       auto joint_pos = std::dist(info_.joints.begin(), found_joint);
       
       used_interface_per_joint_[dist][joint_pos] = true;
    }
  
  return true;
```

File: hardware_interface/src/fake_components/generic_system.cpp (hunk 1 above)
Comment (destogl) [reply hardware_interface/src/fake_components/generic_system.cpp:337]: Also not allowed if velocity is missing and there are only ACC and POS interfaces.

**Review by destogl — COMMENTED** [review]
As per call.

File: hardware_interface/src/fake_components/generic_system.cpp [hunk 3]
```diff
@@ -47,6 +48,18 @@ return_type GenericSystem::configure(const hardware_interface::HardwareInfo & in
     fake_sensor_command_interfaces_ = false;
   }
 
+  // check if there is parameter that disables commands
+  // this way we simulate disconnected driver
+  it = info_.hardware_parameters.find("disable_commands");
+  if (it != info.hardware_parameters.end())
+  {
+    command_propagation_disabled_ = it->second == "true" || it->second == "True";
```
[the remark this answers is not in the document]
Comment (livanov93) [reply hardware_interface/src/fake_components/generic_system.cpp:56]: Still you get the string here. Don't we?

**Pull request closed without merging**